### PR TITLE
fix: Check if $ZDOTDIR exists to find correct path to .zshrc

### PR DESCRIPTION
**Pull request by thales-maciel**
Zsh provides a way to keep all related files in a separate directory by exporting a `ZDOTDIR` variable, but we don't check It.

This is a minor issue, but may lead to a bad first time experience for some users, since the required lines would be added in a `.zshrc` file that is never sourced.

### CODE REVIEW

File: os/env/path-extender-posix/path-extender-posix.spec.ts [hunk 1]
```diff
@@ -222,6 +223,37 @@ export PNPM_HOME="${pnpmHomeDir}"
 export PATH="$PNPM_HOME:$PATH"
 # pnpm end`)
   })
+  it('should target config file in custom directory when ZDOTDIR is present', async () => {
+    const customDir = path.join(homeDir, 'customDir')
+    process.env.ZDOTDIR = customDir
+    fs.mkdirSync(customDir)
+    console.log('customDir', customDir)
```
Comment (zkochan) [inline os/env/path-extender-posix/path-extender-posix.spec.ts:230]: remove console.log

File: os/env/path-extender-posix/path-extender-posix.ts [hunk 2]
```diff
@@ -72,7 +72,13 @@ async function updateShell (
...
 
 async function setupShell (shell: 'bash' | 'zsh', dir: string, opts: AddDirToPosixEnvPathOpts): Promise<PathExtenderPosixReport> {
-  const configFile = path.join(os.homedir(), `.${shell}rc`)
+  const shellConfigName = `.${shell}rc`
+  let configFile!: string
+  if (shell === 'zsh') {
+    configFile = path.join((process.env.ZDOTDIR || os.homedir()), shellConfigName)
+  } else {
+    configFile = path.join(os.homedir(), shellConfigName)
+  }
```
Comment (zkochan) [inline os/env/path-extender-posix/path-extender-posix.ts:81]: create a new function for getting the configFile path

File: os/env/path-extender-posix/path-extender-posix.spec.ts (hunk 1 above)
Comment (thales-maciel) [reply os/env/path-extender-posix/path-extender-posix.spec.ts]: my bad!

File: os/env/path-extender-posix/path-extender-posix.ts (hunk 2 above)
Comment (thales-maciel) [reply os/env/path-extender-posix/path-extender-posix.ts]: done

**Review by zkochan — APPROVED** [review]

**Pull request merged**